terraform/lib/dependabot/terraform/file_parser.rb

```diff
@@ -163,10 +163,30 @@ def provider_source_from(source_address, name)
         ]
       end
 
+      def prefer_lockfile_source(dependency)
+        return dependency if dependency.requirements.count == 1
+
+        lockfile_requirement = dependency.requirements.select{ |requirement| requirement[:source][:type] == "lockfile" }.first
+        provider_requirement = dependency.requirements.select{ |requirement| requirement[:source][:type] == "provider" }.first
+
+        if lockfile_requirement[:source][:registry_hostname] == provider_requirement[:source][:registry_hostname] &&
+           lockfile_requirement[:source][:module_identifier] == lockfile_requirement[:source][:module_identifier]
+          return dependency.requirements.select!{ |requirement| requirement == lockfile_requirement }
+        end
+
+        dependency
+      end
+
       def registry_source_details_from(source_string)
         parts = source_string.split("//").first.split("/")
 
-        if parts.count == 3
+        if parts.count == 2
+          {
+            type: "provider",
```

Comment:
I think this is now handled in a different method https://github.com/dependabot/dependabot-core/blob/0a9377f61e31f3ecac9bf88612504a8fa1f1d260/terraform/lib/dependabot/terraform/file_parser.rb#L89-L110 so I don't think we need to account for it here